daffodil-lib/src/main/scala/org/apache/daffodil/lib/xml/DaffodilXMLLoader.scala

```diff
@@ -172,14 +174,25 @@ class DFDLCatalogResolver private ()
   }
 
   def resolveURI(uri: String): String = {
-    init
     val optURI = resolveCommon(uri, null, null)
     optURI match {
       case None => null
       case Some(uri) => uri.toString
     }
   }
 
+  override def resolve(href: String, base: String): Source = {
+    val optURI = resolveCommon(null, href, base)
+    optURI match {
+      case None => null
```

Comment:
I've added a test to show resolution failing, but it still does not hit this `None` case. Baed on a comment and the code, it looks like `resolveCommon` only returns `None` in certain unit tests. In most other cases it just throws an exception that is handled elsewhere.

https://github.com/apache/daffodil/blob/main/daffodil-lib/src/main/scala/org/apache/daffodil/lib/xml/DaffodilXMLLoader.scala#L250-L254

So I think this None case is just dead code (some other resolve functions have this same dead code, but some do hit it), but is maybe good to keep it just so it is consistent the the other resolve functions or in case we every change the behavior of resolveCommon?

Reply:
I would then suppress the codeCov thing with their suppression comments and change the code to do Assert.invariantFailed(...) on the None return.

Reply:
Done, we now have full coverage of the new resolve function.